core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -1656,6 +1659,15 @@ def save_info(self) -> dict:
         json_path = self.path + self.model_info_json_name
         save_json.save(path=json_path, obj=info)
         return info
+    
+    def _get_maximum_resources(self) -> Dict[str, float]:
+        """
+        Get the maximum resources allowed to use for this model.
+        This can be useful when model not scale well with resources, i.e. cpu cores.
+        Return empty dict if no maximum resources needed
+        """
+        return {}
```

Comment:
This should check for `ag.max_gpus`, `ag.max_cpus`, however we can add this in a follow-up PR.

Ideally user should be able to do:

```
 # Force each fold to use at most 8 CPUs and never use GPU
hyperparameters = {
     'GBM': {'ag.max_cpus': 8, 'ag.max_gpus': 0}
}
```

```
# Force each fold to use at minimum 8 CPUs (for example, if bagging 8 folds with 32 cores, this forces to only do 4 folds in parallel, since each fold needs 8 CPUs)
hyperparameters = {
     'GBM': {'ag.min_cpus': 8}  
}
```